processor/src/main/java/org/mapstruct/ap/internal/model/assignment/AdderWrapper.java

```diff
@@ -39,7 +39,12 @@ public AdderWrapper( Assignment rhs,
         // localVar is iteratorVariable
         String desiredName = Nouns.singularize( adderIteratorName );
         rhs.setSourceLoopVarName( rhs.createUniqueVarName( desiredName ) );
-        adderType = first( getSourceType().determineTypeArguments( Collection.class ) );
+        if ( getSourceType().isCollectionType() ) {
+            adderType = first( getSourceType().determineTypeArguments( Collection.class ) );
+        }
+        else { // array
+            adderType =  getSourceType().getComponentType();
+        }
```

Comment:
This won't work if the source type is iterable i.e. `Iterable<String>`. Can you please add a test case for this as well?
```suggestion
        if ( getSourceType().isArrayType() ) {
            adderType  =  getSourceType().getComponentType();
        }
        else { // iterable
            adderType = first( getSourceType().determineTypeArguments( Collection.class ) );
            adderType =  getSourceType().getComponentType();
        }
```

Reply:
added